### DISCUSSION TITLE
Localized array fields

### DISCUSSION THREAD

**Discussion by wiesson** [payloadcms/payload#757] [Q&A]
I have noticed that array fields are not automatically available in other languages (fall back also doesn't work) - do I need to create a collection (and new relation) to make it work? 

I'd like to have everything localized except the images, but it looks like I have to recreate the whole array for every other language. Example:

```Typescript
{
  name: "itineraries",
  type: "array",
  localized: true,
  labels: {
    singular: "Itinerary",
    plural: "Itineraries",
  },
  fields: [
    {
      name: "title",
      type: "text",
      required: true,
      localized: true,
    },
    {
      name: "duration",
      type: "number",
      localized: false,
    },
    {
      name: "image",
      label: "Media",
      type: "upload",
      relationTo: "media",
      localized: false,
      admin: {
        description:
          "Maximum upload file size: 12MB. Recommended file size for images is <500KB.",
      },
    },
    {
      name: "content",
      type: "textarea",
      localized: true,
    },
    {
      name: "stays",
      type: "array",
      fields: [
        {
          name: "Hotel",
          type: "relationship",
          relationTo: "hotels",
        },
      ],
    },
  ],
},
```

**Comment by denolfe** [comment]
Hey @wiesson , the behavior you are seeing is due to having `localized: true` on the top-level array. Removing that and leaving the others should give you what you are looking for.

**Reply by wiesson** [reply]
Oh, that easy? I guess, I will lose the existing content if I remove it on top level?

**Reply by DanRibbens** [reply]
@wiesson The collections in the DB will have your saved content until a new save is made that modifies the document with the field changes that you make.

The Payload APIs won't show the data since the structure won't match the fields once you change it. You'll need to do some data migration is required to accomdate existing data.

I've mentioned some 3rd party migration tools in other discussions and I'm waiting to get a chance to try some of them and document the workflow.

Would like to hear if anyone gets a chance to use one of these or an alternative?
[migrate-mongoose](https://www.npmjs.com/package/migrate-mongoose)
[migrate-mongo](https://www.npmjs.com/package/migrate-mongo)
